### PR TITLE
Finalise api 0.3

### PR DESCRIPTION
**Pull request by benjaminbollen**
[Review on Reviewable](https://reviewable.io/reviews/maidsafe/routing/613)

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @inetic

(maidsafe_highfive has picked a reviewer for you, use r? to override)

File: examples/simple_key_value_store.rs [hunk 1]
```diff
@@ -280,7 +278,8 @@ struct Client {
 impl Client {
     fn new(_bootstrap_peers: Vec<Endpoint>) -> Result<Client, RoutingError> {
```
Comment (dirvine) [inline examples/simple_key_value_store.rs]: Are these bootstrap things not completely take care of in crust ? i.e. via the config file.

File: src/error.rs [hunk 2]
```diff
@@ -30,20 +30,25 @@ use data::Data;
...
 impl error::Error for ResponseError {
     fn description(&self) -> &str {
         match *self {
-            ResponseError::NoData => "No Data",
-            ResponseError::InvalidRequest => "Invalid request",
-            ResponseError::FailedToStoreData(_) => "Failed to store data",
+            ResponseError::Abort => "Abort",
+            ResponseError::InvalidRequest(_) => "Invalid request",
+            ResponseError::FailedRequestForData(_) => "Failed request for data",
+            ResponseError::HadToClearSacrificial(size) => "Had to clear {:?} bytes of Sacrificial
```
Comment (dirvine) [inline src/error.rs]: I am not sure this should be even known to routing.

File: examples/simple_key_value_store.rs (hunk 1 above)
Comment (benjaminbollen) [inline examples/simple_key_value_store.rs]: yes they are, and I have another WIP (not yet opened PR, where Im taking that out of the example)  good spotting !

File: src/error.rs (hunk 2 above)
Comment (benjaminbollen) [inline src/error.rs]: I agree!  I would want to minimise what Routing knows/defines about ExternalRequest ExternalResponse and associated errors; it's stupid that the user can only pass these error types - as one example

Comment (benjaminbollen) [inline src/error.rs]: Is there a quick way to fix it now, or do we mark that for the technical death sprint?

**Pull request merged**